tests/assertions.js

```diff
@@ -59,6 +59,41 @@ export function shouldBeIssuedVc({issuedVc}) {
   }
 }
 
+/**
+ * Some issuers validate credentials before issuing and others don't.
+ * This asserts that a negative case is rejected at either issuance
+ * or verification.
+ *
+ * @param {object} options - Options to use.
+ * @param {object}options.endpoints - An implementer's endpoints.
+ * @param {object} options.negativeTest - An invalid credential for issuance.
+ * @param {string} options.reason - The reason the negativeTest should fail.
+ *
+ */
+export async function shouldRejectEitherIssueOrVerify({
+  endpoints,
+  negativeTest,
+  reason
+}) {
+  let error;
+  let result;
+  try {
+    //depending on the issuer this may fail to issue an invalid VC
+    result = await endpoints.issue(negativeTest);
+  } catch(e) {
+    error = e;
+  }
+  // if an issuer fails to issue a VC with invalid validFrom
+  // and/or validUntil we count this as a success and return early
+  if(error) {
+    return;
```

Comment:
Isn't this just going to swallow the issuer error here?

Reply:
it will, you are correct, but we don't have a way of asserting on that error to ensure it is the correct error and _as this is a negative test, it should pass if the issuer errors_. We could `console.warn` the error or we could at least check the response code. The later is something I have been working on here: https://github.com/w3c-ccg/data-integrity-test-suite-assertion/blob/1cd6ee30152272f1bf00e0da8f642eb3593e41e2/assertions.js#L149-L170 One other thing I noticed in this test: Digital Bazaar passed it intermittently and when DB did not pass this test it was due to issues unrelated to the issuer throwing an error. I think it might have been a timeout. So maybe we can improve this test by only not returning the error if the status code is 400 or 422? I'm not sure what status code having an invalid `validFrom` `validUntil` combo should throw.

Reply:
@BigBlueHat perhaps you would prefer that I return the error so it could be asserted on at another time?

Reply:
@BigBlueHat now returns the `{result, error}` provided the assertion does not throw: https://github.com/w3c/vc-data-model-2.0-test-suite/pull/96/commits/c47634e681e83150644f81504203a7a66876368e

Reply:
My greater (and growing) concern here is how dependent that test is on having or understanding _an API_ when these are supposed to be tests about the data model. The lines are getting blurry in test like these as the issuance/verification could fail (or not) for a whole host of unrelated reasons.

What you've changed here is an improvement on what was there, but I'm still not confident that this will test what we think/intend it to test. 😕

Reply:
@BigBlueHat yeah but now we have the ability to assert on the error and any response data in the error. p.s. we should probably start to assert that error responses should return json.